src/Tribe/Utils/Post_Root_Pool.php

```diff
@@ -0,0 +1,164 @@
+<?php
+
+
+class Tribe__Utils__Post_Root_Pool {
+
+	/**
+	 * @var string
+	 */
+	protected $pool_transient_name = 'tribe_ticket_prefix_pool';
+
+	/**
+	 * @var array|bool
+	 */
+	protected static $prefix_pool = false;
+
+	/**
+	 * @var string
+	 */
+	protected $root_separator = '-';
+
+	/**
+	 * @var array
+	 */
+	protected $postfix = 1;
+
+	/**
+	 * Generates a unique root for a post using its post_name.
+	 *
+	 * @param WP_Post $post
+	 *
+	 * @return string
+	 */
+	public function generate_unique_root( WP_Post $post ) {
```

Comment:
Is it possible that this method will be called twice for the _same_ post? If not, ignore this comment!

... but if so, wouldn't we end up with multiple unique roots for the same post? It's clear from the tests what we expect when different posts of the same name are passed in, but not what we expect if the same post is passed in more than once (if indeed it matters).